src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics/SatelliteDynamics.cpp

```diff
@@ -160,63 +165,47 @@ void SatelliteDynamics::DynamicalEquations(const Dynamics::StateVector& x, Dynam
         totalGravitationalAcceleration_SI += gravitationalAcceleration.inFrame(gcrfSPtr_, currentInstant).getValue();
     }
 
-    // Integrate position and velocity states
+    dxdt[3] += totalGravitationalAcceleration_SI[0];
+    dxdt[4] += totalGravitationalAcceleration_SI[1];
+    dxdt[5] += totalGravitationalAcceleration_SI[2];
+
+    // DRAG
+
+    const Real mass = satelliteSystem_.getMass().inKilograms();
+    const Real dragCoefficient = satelliteSystem_.getDragCoefficient();
+    const Real surfaceArea = satelliteSystem_.getCrossSectionalSurfaceArea();
+
+    for (const auto& objectName : environment_.getObjectNames())
+    {
+        // TBI: currently only defined for Earth
+        if (environment_.accessCelestialObjectWithName(objectName)->accessAtmosphericModel() != nullptr)
+        {
+            const Real atmosphericDensity = environment_.accessCelestialObjectWithName(objectName)
```

Comment:
Calling `environment_.accessCelestialObjectWithName` being "costly", let's bind a variable before using it in multiple places?